### PR TITLE
chore: proper VidScheme constructor

### LINKED ISSUE #2659: [TECH_DEBT] - Proper `VidScheme` constructor
### What is this task and why do we need to work on it?

See discussion at https://github.com/EspressoSystems/espresso-sequencer/issues/1047 . All downstream code will use a `VidScheme` constructor that lives here in the hotshot repo.

### What work will need to be done to complete this task?

- hotshot repo pass tests
- downstream repos pass tests. Need to test downstream repos locally before closing this issue.

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

- hotshot repo passes tests with the new constructor
- downstream repos (eg. espresso-sequencer, hotshot-query-service, etc) pass tests with the new constructor

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by ggutoski**
Closes #2659 

### This PR: 

- New file `crates/types/src/vid.rs` with new VID constructor function `vid_scheme`.
- Move a few VID-related things from `data.rs` to `vid.rs`.
- Fix some downstream build warnings. (Our CI currently runs with `--all-features` but some subsets of features throw warnings that are not detected by CI.)

### This PR does not: 

- Need to update downstream repos `hotshot-query-service`, `espresso-sequencer`.
- I got `hotshot-query-service` to build locally under the present hotshot PR, but there will be merge conflicts with https://github.com/EspressoSystems/hotshot-query-service/pull/441
- I did not test in `espresso-sequencer` due to time constraints.

### Key places to review: 

`crates/types/src/vid.rs`. Most other changes are mere consequences of this new file.

### CODE REVIEW

File: crates/types/src/vid.rs
Comment (jbearer) [inline crates/types/src/vid.rs:85]: Nit: other than here, HotShot doesn't know about namespaces at all (and we're even mostly removing its knowledge of transactions). It might make more sense to still call these `{Large,Small}RangeProof`, just without the type parameters.

**Review by jbearer — COMMENTED** [review]
Looks good, except one question I had about the SRS.

Once this is merged, I think I can incorporate it into the query service as part of https://github.com/EspressoSystems/hotshot-query-service/pull/441. Should clean up that PR a lot. I'll leave it to you to incorporate in the sequencer.

File: crates/types/src/vid.rs
Comment (ggutoski) [reply crates/types/src/vid.rs:53]: `test_rng` is deterministic, so this should produce the same SRS each time. We've been doing it like this since the dawn of time so if it were a problem then we'd know about it by now. That said, we definitely need to stop regenerating the SRS for each VID scheme. 😬 This is something on the crypto team's plate. Related: https://github.com/EspressoSystems/HotShot/issues/1686

File: crates/types/src/vid.rs
Comment (ggutoski) [reply crates/types/src/vid.rs]: Ooh good catch thank you. Fixed in https://github.com/EspressoSystems/HotShot/pull/2670/commits/195edec572bb013552280d1a3f6371a490d8251f

**Review by jbearer — APPROVED** [review]

**Pull request merged**
